src/ui-shell/header/header-item.component.ts

```diff
@@ -15,14 +15,24 @@ import { Router } from "@angular/router";
 	selector: "ibm-header-item",
 	template: `
 		<li style="height: 100%">
-			<a
-				class="bx--header__menu-item"
-				role="menuitem"
-				tabindex="0"
-				[href]="href"
-				(click)="navigate($event)">
-				<ng-content></ng-content>
-			</a>
+			<ng-container [ngSwitch]="useRouter">
+				<a *ngSwitchCase="false"
+				   class="bx--header__menu-item"
+				   role="menuitem"
+				   tabindex="0"
+				   [href]="href"
+				   (click)="navigate($event)">
+					<ng-content></ng-content>
+				</a>
+				<a *ngSwitchCase="true"
+				   class="bx--header__menu-item"
+				   role="menuitem"
+				   tabindex="0"
+				   [routerLink]="route"
+				   [routerLinkActive]="activeLinkClass">
```

Comment:
What are the differences between this approach and directly invoking `this.router.navigate`? If they're effectively the same, or if this way is the same but better, I would say we just switch over to this approach entirely

Reply:
The biggest difference is that it supports right click and open in new tab and doesn't send users to a blank void page when doing it while maintaining router navigation and adds active route class support.

Reply:
Gotcha 👍 I'm assuming `routerLink` lets us use raw `href`s as well?

If so it sounds like both approaches are functionally identical, but using `routerLink` directly gives us better UX and more features ... in that case - as long as it's possible to keep the `navigation` event for compat reasons - I would say we just switch over to directly using `routerLink`

Reply:
> Gotcha 👍 I'm assuming `routerLink` lets us use raw `href`s as well?

`routerLink` doesn't accept urls you'd normally put in `href`s but you can use one or the other on the `a` tag.
I would use something like this
```ts
ngOnInit() {
	if (this.route && typeof this._href !== 'undefined') {
		throw new Error('You must only pass one of "route" or "href"')
	}
}
```
to restrict the use to one or the other. In the end though, this requires template switching as having the `routerLink` directive on an anchor tag prevents the `href` attribute from working normally. Unfortunately passing null to the directive doesn't keep it from binding to the element.

I see now why it's currently written the way it is since it's a bit more of an elegant approach, code-wise, albeit rigid in it's behavior.

> If so it sounds like both approaches are functionally identical, but using `routerLink` gives us better UX and more features ... in that case - as long as it's possible to keep the `navigation` event for compat reasons - I would say we just switch over to `routerLink` and drop the other method as much as possible

Keeping the `navigation` event is kind of tricky. I couldn't really emulate the same results as it was doing. I had eventually drilled down to a router events subscription like so
```ts
const subscription = this.router.events
	.pipe(filter(value => value instanceof NavigationEnd))
	.pipe(throttle((value: NavigationEnd) => interval(10000)))
	.pipe(filter((value: NavigationEnd) => value.url.replace(/^\//, '') === this.route[0]))
	.pipe(tap(event => console.log(event)))
	.pipe(mapTo(true))
	.toPromise();
this.navigation.emit(subscription);
```
but it was sending back multiples instead of just a single event.

I'm not sure of the use cases behind it since it is pretty limited and only responds true or false to the route you go to. If I needed to hook into the router events I'm more likely to subscribe to the `router.events` observable as I would have more data to work with and more flexibility.

Reply:
Ah, so yeah, we've definitely coded ourselves into a bit of a box here then ... We can work with this for now, and figure out which method to promote and which to deprecate down the line ... There's a lot of API overlap so that does make things simpler, and `useRouter` keeps this from breaking anyone